Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOProjectCardViewModel.swift

```diff
@@ -0,0 +1,177 @@
+import Combine
+import Foundation
+import KsApi
+
+protocol PPOProjectCardViewModelInputs {
+  func sendCreatorMessage()
+  func performAction(action: PPOProjectCardViewModel.Action)
+}
+
+protocol PPOProjectCardViewModelOutputs {
+  var sendMessageTapped: AnyPublisher<Void, Never> { get }
+  var actionPerformed: AnyPublisher<PPOProjectCardViewModel.Action, Never> { get }
+}
+
+typealias PPOProjectCardViewModelType = Identifiable & ObservableObject & PPOProjectCardViewModelInputs &
+  PPOProjectCardViewModelOutputs
+
+final class PPOProjectCardViewModel: PPOProjectCardViewModelType {
+  internal private(set) var isUnread: Bool
+  internal private(set) var alerts: [Alert]
+  internal private(set) var imageURL: URL
+  internal private(set) var title: String
+  internal private(set) var pledge: GraphAPI.MoneyFragment
+  internal private(set) var creatorName: String
+  internal private(set) var address: String?
+  internal private(set) var actions: (Action, Action?)
+  internal private(set) var parentSize: CGSize
```

Comment:
Using vars here instead of subjects is inconsistent with how we usually do things in this codebase. I don't personally have a strong opinion about one or the other, so I like to go with whatever we usually do. However, if you think that using vars simplifies things or makes things better in a way that's more valuable than keeping the patterns we usually use, I'm fine with leaving this as it is.

Reply:
I've updated these in 80b1694 to be Publisher-based, they don't actually need to change yet so the vars were premature, but they are reactive now, and if that changes later then we can replace that in the view model and the view should update without issue.